### ISSUE TITLE
Easy way to print hex for raw tx hex?

### ISSUE THREAD

**Issue by sanket1729** [rust-bitcoin/rust-bitcoin#1706]
Perhaps, I am mis-using the library. I have to do the following to print tx-hex which is non-intuitive to new users(and even me).

```rust
		bitcoin::consensus::encode::serialize_hex(&tx);
```

**Comment by apoelstra** [comment]
We could probably implement `LowerHex` and `UpperHex` for everything that implements `consensus::Encodable`. I don't know that this would be any more discoverable.

What would an intuitive API look like to you?

**Comment by Kixunil** [comment]
Not sure if *anything* but `Transaction` definitely yes. Hex encoding is very common for transactions.

**Comment by dpc** [comment]
> What would an intuitive API look like to you?

In Fedimint we have [`consensus_encode_to_vec`](https://github.com/fedimint/fedimint/blob/60adf3717a104e24956a4b5efcd4c5df35552f7d/fedimint-core/src/encoding/mod.rs#L68), and would make sense to have `consensus_encode_to_hex`.

**Comment by Kixunil** [comment]
`LowerHex`/`UpperHex` is definitely more idiomatic. We can also probably reuse work in #1254

**Comment by Ademan** [comment]
I had trouble finding serialize_hex at all, @apoelstra had to look it up and encouraged me to ack the related issue. (Hopefully I found the right one)

**Comment by tcharding** [comment]
Thanks @Ademan, improving the hex situation is top of the list atm. (ref: https://github.com/rust-bitcoin/rust-bitcoin/pull/1531#issuecomment-1869004643)

**Comment by tcharding** [comment]
This can potentially be closed by either #2318 or #2039

**Comment by Kixunil** [comment]
Closed by #2039

**Comment by sanket1729** [comment]
Is this resolved? I still don't find a method of Transaction struct to serialize raw transaction? Are we missing  an implementation of `DisplayHex` for `&Transaction` .

Feel free to close this issue if I am missing something

**Comment by sanket1729** [comment]
cc @apoelstra @tcharding incase this notification gets lost

**Comment by apoelstra** [comment]
Agreed, I dont' think it's resolved. We should implement `Display`/`LowerHex`/`UpperHex` for `Transaction` and `DisplayHex` for `&Transaction`.

**Comment by tcharding** [comment]
I went to do this a month or so ago and decided not to because the code lives in `rust-bitcoin` (ATM anyway) and I was focusing on the leaf crates.
